### ISSUE TITLE
md-datepicker doesn't change model value if date entered manually

### ISSUE THREAD

**Issue by vangraan** [angular/material#6125]
ng-model only get updated when a value is selected from the calendar popup. If you change the date by entering text only without selecting a date from the popup the bound value will not change.

**Comment by jayeshanandani** [comment]
I think you need to take a look at parseDate which is provided by $mdDateLocaleProvider. It helps to convert user entered input into a date object and set it on datepicker.

Hope that helps.

**Comment by lwj5** [comment]
@jayeshanandani The parseDate function does update the datepicker but does not update the input field. the is similar to the result I had (even when moment.js is used). Please correct me if I'm wrong.

I think what @vangraan is refering to is the value displayed in the input field after blur or a certain debounce time. Would be useful if the it calls formatDate on parseDate's return to update the input/model value.

**Comment by vangraan** [comment]
I have created a example here. This example works correctly in chrome (My Date format in chrome is automatically set to a different format to my PC) But when I then run the same example it won't work on Edge
http://codepen.io/anon/pen/LGEwRm?editors=100

@jayeshanandani and @lwj5 I then looked at your suggestions and created the following example which will work on both browsers. It uses formatDate.
http://codepen.io/anon/pen/vNLyGJ?editors=101

Thanks for the help!
